src/map.cpp

```diff
@@ -4472,17 +4472,21 @@ item &map::add_item( const tripoint &p, item new_item )
 item map::water_from( const tripoint &p )
 {
     if( has_flag( "SALT_WATER", p ) ) {
-        return item( "salt_water", 0, item::INFINITE_CHARGES );
+        item ret( "salt_water", 0, item::INFINITE_CHARGES );
+        ret.set_item_temperature( std::max( temp_to_kelvin( g->weather.get_temperature( p ) ), 277.15 ) );
```

Comment:
`277.15` is used in three more places now, so it would be nice to extract it to static constexpr expression.

Reply:
`constexpr int cold = 40; // ~4.4 C`  happens to be that already so we can use that.

Reply:
int doesn't match double, some typecasting needed?